libdnf5/conf/option_string_list.cpp

```diff
@@ -197,35 +197,70 @@ void OptionStringContainer<T, IsAppend>::test(const ValueType & value) const {
     }
 }
 
-// Since strtok_r modifies its str input we copy `value` param
 template <typename T, bool IsAppend>
 T OptionStringContainer<T, IsAppend>::from_string(std::string value) const {
-    ValueType tmp;
-    char * str = nullptr;
-    char * token = nullptr;
-    char * saveptr = nullptr;
-
-    // If the first token in the string is empty (we start with a delimiter) it's
-    // a special case, it can be used to clear existing content of the option.
-    auto start = value.find_first_not_of(' ');
-    if (start != std::string::npos && strchr(get_delimiters(), value[start]) != nullptr) {
-        tmp.insert(tmp.end(), "");
-    }
+    ValueType ret;  // Container to hold the resulting parsed items
+
+    const std::string_view delimiters{get_delimiters()};
+
+    bool item_started = false;  // True after first non-space or escaped character in current item
+    size_t valid_item_len = 0;  // Length of current item excluding trailing unescaped spaces
+    bool escape = false;        // True when backslash was encountered (next char is escaped)
+    std::string item;           // Current item being built character by character
+
+    const auto end_it = value.end();
+    for (auto it = value.begin(); it != end_it; ++it) {
+        // Handle escaped character: always add it
+        if (escape) {
+            escape = false;
+            item += *it;
+            valid_item_len = item.size();
+            item_started = true;
+            continue;
+        }
 
-    for (str = value.data();; str = nullptr) {
-        token = strtok_r(str, get_delimiters(), &saveptr);
-        if (token == nullptr) {
-            break;
+        if (*it == '\\') {
+            escape = true;
+            continue;
+        }
```

Comment:
From the doc string it is not clear, how the parser behaves if it encounters a backslash not followed by another backslash or delimiter. Special case - trailing backslash. I believe currently it is ignored: `from_string("fir\st,second\") => ["first", "second"]`.
I think the current implementation is valid, but it should be mentioned in the documentation.